Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/workspace/FindSymbolsCommandExecutor.kt

```diff
@@ -4,25 +4,34 @@ import com.intellij.openapi.editor.Editor
 import com.intellij.openapi.project.Project
 import com.tabnineCommon.chat.commandHandlers.utils.StringCaseConverter
 import com.tabnineCommon.chat.commandHandlers.utils.SymbolsResolver
-import java.util.stream.Stream
-import kotlin.streams.toList
+import com.tabnineCommon.chat.commandHandlers.utils.submitReadAction
+import java.util.concurrent.CompletableFuture
 
 private const val MAX_RESULTS_PER_SYMBOL = 5
 
 class FindSymbolsCommandExecutor : CommandsExecutor {
     override fun execute(arg: String, editor: Editor, project: Project): List<String> {
         val camelCaseArg = StringCaseConverter.toCamelCase(arg)
         val snakeCaseArg = StringCaseConverter.toSnakeCase(arg)
-        val camelCaseSymbols = SymbolsResolver.resolveSymbols(
-            project, editor.document, camelCaseArg,
-            MAX_RESULTS_PER_SYMBOL
-        )
-        val snakeCaseSymbols = SymbolsResolver.resolveSymbols(
-            project, editor.document, snakeCaseArg,
-            MAX_RESULTS_PER_SYMBOL
+
+        val tasks = listOf(
+            submitReadAction {
+                SymbolsResolver.resolveSymbols(
+                    project, editor.document, camelCaseArg,
+                    MAX_RESULTS_PER_SYMBOL
+                )
+            },
+            submitReadAction {
+                SymbolsResolver.resolveSymbols(
+                    project, editor.document, snakeCaseArg,
+                    MAX_RESULTS_PER_SYMBOL
+                )
+            }
         )
 
-        return Stream.concat(camelCaseSymbols.stream(), snakeCaseSymbols.stream())
+        CompletableFuture.allOf(*tasks.toTypedArray()).get()
```

Comment:
im not putting a TO on this awaiting, since the upper layer already caps this operation with a TO (2.5s).